cli/flags.rs

```diff
@@ -581,735 +810,502 @@ Demonstrates breaking the input up by space delimiter instead of by lines:
-              .values_of("")
-              .unwrap()
-              .map(String::from)
-              .collect();
-
-            let (script_args, flags_) = parse_script_args(script_args, flags);
-            flags = flags_;
-            argv.extend(script_args);
-          }
-          DenoSubcommand::Run
-        }
-        _ => unreachable!(),
-      }
+fn reload_arg_parse(flags: &mut DenoFlags, matches: &ArgMatches) {
+  if matches.is_present("reload") {
+    if matches.value_of("reload").is_some() {
+      let cache_bl = matches.values_of("reload").unwrap();
+      let raw_cache_blacklist: Vec<String> =
+        cache_bl.map(std::string::ToString::to_string).collect();
+      flags.cache_blacklist = resolve_urls(raw_cache_blacklist);
+      debug!("cache blacklist: {:#?}", &flags.cache_blacklist);
+      flags.reload = false;
+    } else {
+      flags.reload = true;
     }
-    ("xeval", Some(xeval_match)) => {
-      flags.allow_net = true;
-      flags.allow_env = true;
-      flags.allow_run = true;
-      flags.allow_read = true;
-      flags.allow_write = true;
-      flags.allow_hrtime = true;
-      argv.push(XEVAL_URL.to_string());
-
-      if xeval_match.is_present("delim") {
-        let delim = xeval_match.value_of("delim").unwrap();
-        argv.push("--delim".to_string());
-        argv.push(delim.to_string());
-      }
+  }
+}
 
-      if xeval_match.is_present("replvar") {
-        let replvar = xeval_match.value_of("replvar").unwrap();
-        argv.push("--replvar".to_string());
-        argv.push(replvar.to_string());
-      }
+fn importmap_arg<'a, 'b>() -> Arg<'a, 'b> {
+  Arg::with_name("importmap")
+    .long("importmap")
+    .value_name("FILE")
+    .help("Load import map file")
+    .long_help(
+      "Load import map file
+Docs: https://deno.land/std/manual.md#import-maps
+Specification: https://wicg.github.io/import-maps/
+Examples: https://github.com/WICG/import-maps#the-import-map",
+    )
+    .takes_value(true)
+}
 
-      let code: &str = xeval_match.value_of("code").unwrap();
-      argv.push(code.to_string());
+fn importmap_arg_parse(flags: &mut DenoFlags, matches: &clap::ArgMatches) {
+  flags.import_map_path = matches.value_of("importmap").map(ToOwned::to_owned);
+}
 
-      DenoSubcommand::Run
-    }
-    (script, Some(script_match)) => {
-      argv.extend(vec![script.to_string()]);
-      // check if there are any extra arguments that should
-      // be passed to script
-      if script_match.is_present("") {
-        let script_args: Vec<String> = script_match
-          .values_of("")
-          .unwrap()
-          .map(String::from)
-          .collect();
-
-        let (script_args, flags_) = parse_script_args(script_args, flags);
-        flags = flags_;
-        argv.extend(script_args);
-      }
-      DenoSubcommand::Run
+// TODO move this to utility module...
+// TODO what does this function even do?
+pub fn resolve_urls(urls: Vec<String>) -> Vec<String> {
+  use url::Url;
```

Comment:
Yeah, this is very specific to cache blacklist - generally it strips `#fragment` part of URL. I'm not sure though why it pops last slash /shrug

Reply:
Thanks - added a little doc line.